apps/dashr-ideogram/app.R

```diff
@@ -0,0 +1,1162 @@
+                id = 'ideogram-feature-view-options'
+              )
+            ))
+          )
+        ))
+      )),
+      
+      
+  
+      
+      htmlDiv(do.call(dashbioIdeogram, ideograms_initial[['custom']]), style = list('display' = 'none')),
+      
+      
+      
+      dccStore(id='ideo-custom-data', data=ideograms_initial[['custom']]),
+      dccStore(id='ideo-homology-data', data=ideograms_initial[['homology']]),
+      dccStore(id='brush-ideo-data', data=ideograms_initial[['brush']]),    # Might be brush-ideo-data
+      dccStore(id='ideo-annotations-data', data=ideograms_initial[['annotations']])
+    )
+    )
+  ))
+  
+  header <- htmlDiv(children = list(
+    htmlSpan(list(htmlA(
+      htmlImg(src='assets/index.png', height = '50', width = '50',
+              style = list('top' = '10', 'margin-left' = '10px')),
+      href='https://dash-bio.plotly.host/Portal/'))),
+    
+      "    Dash Ideogram",
+  
+      
+      htmlA(href="http://github.com/plotly/dash-bio/blob/master/tests/dashbio_demos/app_ideogram.py",
+            style = list("color" = "white",
+                         "border" = "1px solid white",
+                         "font-size" = "18px",
+                         "position" = "relative",
+                         "float" = "right",
+                         "padding-top" = "10px",
+                         "padding-left" = "20px",
+                         "padding-right" = "20px",
+                         "padding-bottom" = "10px",
+                         "font-weight" = "100",
+                         "margin-right" = "30px",
+                         "top" = "17px",
+                         "text-decoration" = "none"
+                         
+                         ),
+            children = "View on Github"),
+    
+    
+    htmlImg(src = "assets/github.png", width = '50', height = '50', style = list(
+      "position" = "relative", "float" = "right",
+      "margin-right" = "10px",
+      "top" = "17px"
+    ))
+    ), className = 'ideogramheader')
+  
+  
+  
+  
+  app$layout(htmlDiv(list(
+      header,
+      ideolayout,
+      htmlDiv(list(
+      "                "
+      ), className = "blankspace"),
+      daqToggleSwitch(
+        id = 'daq-light-dark-theme',
+        color = '#FDFCFF',
+        labelPosition = 'bottom',
+        label = list('Light', 'Dark'),
+        style = list('width' = '250px', 'margin' = 'auto', 'display' = 'none'),
+        value = FALSE
+      )
+    ), style = list('background-color' = '#FDFCFF')
+  ))
+  
+  
+  
+  #####################################################################################################
+  
+  # Callbacks
+  
+  
+  
+  # Select which chromosomes to show for the customizable feature.
+  
+   app$callback(
+      output = list(id = 'ideo-custom', property = 'chromosomes'),
+      params = list(input(id = 'displayed-chromosomes', property = 'value')),
+  
+      update_ideogram <- function(value) {

```

Comment:
Interesting. Why an identity function here? Is this just to keep the same structure as in the other callbacks?

Reply:
Pretty much, just so it's a bit easier for someone who might be new to callbacks.